### ISSUE TITLE
SourceGenerators using RegisterPostInitializationOutput break Wpf projects

### ISSUE THREAD

**Issue by tom-englert** [dotnet/roslyn#63489]
**Version Used**: 
Microsoft Visual Studio Professional 2022 (64-bit) - Preview
Version 17.4.0 Preview 1.0

**Steps to Reproduce**:
1. Have a simple library project that consumes a source generator providing sources via `RegisterPostInitializationOutput` that works fine
2. Target multiple frameworks (e.g. `<TargetFrameworks>netcoreapp3.1;net5.0-windows;net6.0-windows</TargetFrameworks>`)
3. Change the project to a WPF project (`<UseWPF>true</UseWPF>`)
4. Add an empty XAML element, e.g. a `UserControl`

**Expected Behavior**:
Project compiles after each step

**Actual Behavior**:
After step 4. the project does no longer compile. It complaints that all the generated code is duplicated N times, where N is the number of target frameworks.

**Comment by tom-englert** [comment]
I have picked some arbitrary project from GH and added the steps to reproduce this issue:
https://github.com/tom-englert/dunet/tree/dotnet/roslyn/issues/63489

**Comment by tom-englert** [comment]
Looks like this is a duplicate of #62711
